components/sup/src/manager/service/mod.rs

```diff
@@ -732,11 +706,6 @@ impl Service {
         Ok(())
     }
 
-    fn run_hook(&mut self, hook: HookType) -> Result<()> {
-        self.hooks.compile(&self.service_group, &self.config);
-        self.hooks.try_run(hook, &self.service_group)
-    }
-
     /// Update our own service rumor with a new configuration from the packages exported
```

Comment:
Yeah, I was just hitting it hard with a hammer when I was making sure things were compiled on call. It changed the signatures of the functions to need to be mutable and made the API worse in the end. I'm pretty sure having the hooks compile in the two spots we are now is all the places it needs to be called in. We can add additional places at no cost because the caching is still in place when calling compile